aws-xray-recorder-sdk-benchmark/tst/main/java/com/amazonaws/xray/entities/EntityBenchmark.java

```diff
@@ -69,57 +78,42 @@ public void doTearDown() {
 
     // Construct a segment
     @Benchmark
-    @BenchmarkMode(Mode.All)
-    @Fork(value=1)
-    @Warmup(iterations = 20)
-    @Measurement(iterations = 20)
-    @OutputTimeUnit(TimeUnit.NANOSECONDS)
     public Segment constructSegmentBenchmark(BenchmarkState state) {
```

Comment:
While we're cleaning, can we look for benchmarks using `Level.Invocation` and remove them? They're not real benchmarks for us since it's designed for much longer-running invocations (guessing when the benchmarks were added someone didn't know about the dragons).

http://javadox.com/org.openjdk.jmh/jmh-core/1.7/org/openjdk/jmh/annotations/Level.html#Invocation

Reply:
I can definitely remove them, looks like `Level.Iteration` is more appropriate. Don't want to mess with any dragons.

Reply:
Actually, it turns out some of the tests need a per-invocation setup, e.g. to put a segment in context then end it, or clear segment context before beginning a new segment.

So we can either keep the necessary invocation rules, or scrap the benchmarks that require them since we already have full lifecycle benchmarks (e.g. only have `beginEndSegment` as opposed to having `beginSegment` and `endSegment` benchmarks). I'm in favor of the latter since some of the benchmarks can be quite redundant.

What do you think?

Reply:
Discussed offline, will remove the non-lifecycle Recorder tests since it's very possible that they are not accurate measurements. Using a workaround with `OperationsPerInvocation` for entity benchmarks.